### ISSUE TITLE
A question about training code

### ISSUE THREAD

**Issue by stillwaterman** [davidtellez/contrastive-predictive-coding#6]
Hi, nice code, but I'm confused about the training code of encoder. I think the training of encoder via CPC is  unsupervised, however, you used some label information (made by original mnist dataset), which makes the training process supervised. Am I right?

**Comment by GeoloeG-IsT** [comment]
Yes, I have the same feeling, and I had the same feeling in the paper itself, can someone clarifies?

**Comment by immersive3d** [comment]
The labels information was used to organize data into positive and negative examples. The actual contrastive prediction was on images. No labels were used there.
